packages/training/Dockerfile

```
@@ -0,0 +1,75 @@
+    && rm -rf /var/lib/apt/lists/*
+
+# Create app directory
+WORKDIR /app
+
+# Install Python dependencies
+COPY python/requirements.txt ./requirements.txt
+RUN python3.11 -m pip install --no-cache-dir --upgrade pip && \
+    python3.11 -m pip install --no-cache-dir -r requirements.txt
+
+# Install vLLM (separate layer for caching)
+RUN python3.11 -m pip install --no-cache-dir vllm>=0.4.0
```

Hadolint (2.14.0)

[warning] Pin versions in pip. Instead of `pip install <package>` use `pip install <package>==<version>` or `pip install --requirement <requirements file>`

(DL3013)